lib/pbench/cli/server/tree_manage.py

```diff
@@ -1,14 +1,107 @@
+from datetime import datetime, timedelta, timezone
+import errno
+import fcntl
+from logging import Logger
+
 import click
 
 from pbench.cli import pass_cli_context
 from pbench.cli.server import config_setup
 from pbench.cli.server.options import common_options
 from pbench.common.logger import get_pbench_logger
-from pbench.server import BadConfig
+from pbench.server import BadConfig, OperationCode
 from pbench.server.cache_manager import CacheManager
+from pbench.server.database.models.audit import Audit, AuditStatus, AuditType
+
+# Length of time in hours to retain unreferenced cached results data.
+# TODO: this could become a configurable setting?
+CACHE_LIFETIME = 4.0
+
+
+def reclaim_cache(tree: CacheManager, logger: Logger, lifetime: float = CACHE_LIFETIME):
+    """Reclaim unused caches
+
+    Args:
+        tree: the cache manager instance
+        lifetime: number of hours to retain unused cache data
+        logger: a Logger object
+    """
+    window = datetime.now(timezone.utc) - timedelta(hours=lifetime)
+    total_count = 0
+    has_cache = 0
+    reclaimed = 0
+    reclaim_failed = 0
+    for tarball in tree.datasets.values():
+        total_count += 1
+        if tarball.unpacked:
+            has_cache += 1
+            date = datetime.fromtimestamp(
+                tarball.last_ref.stat().st_mtime, timezone.utc
+            )
+            if date < window:
+                logger.info(
```

Comment:
Yes, this is a bit "chatty" ... but not compared to the 0.69-11 dispatcher and culler! I'm submitting these logs basically so we can get an idea through ops reviews of how cache management is going. Some of these probably can be removed "eventually" when we're comfortable with the day-to-day operation.

(I have issues with the idea of auditing cache management, too; but, again, in these early stages, and especially since we're unpacking and retaining the entire tree, it seemed worthwhile.)

Reply:
The idea was to report that it had been selected, just to check. Again, I'd expect to go back in and remove most of the noise after an ops review or two shows that it seems to be working as expected.